### DISCUSSION TITLE
Retry Logic-Duplicating the test case count in allure report

### DISCUSSION THREAD

**Discussion by KarthikaGokulakrishnan** [allure-framework/allure2#2510] [Questions & Support] [answered]
****Question:****
In my retry configuration if the run mode is 0, In allure report it displays the correct test case count(3). If I kept run mode as 1 2 or 3 while executing it attempting for the respective count but for each retry new test result JSON is generated in allure-results.So test case count is duplicated as 6 in allure report. I want last attempted execution to be generated in the overview and retried should be in retry tab allure report. No duplicate count should get displayed. How to do that?

This is my cypress.config
const { defineConfig } = require("cypress");
const allureWriter = require("@shelex/cypress-allure-plugin/writer");
const junitReporter = require('mocha-junit-reporter');
const { allureCypress } = require("allure-cypress/reporter");
const fs = require('fs');

module.exports = defineConfig({
  experimentalModifyObstructiveThirdPartyCode: true,
  chromeWebSecurity: false,
  
  env: {
    allure: true,
    allureReuseAfterSpec: true,
    grepFilterSpecs: true,
    allureAddVideoOnPass: false,
    allureResultsPath: "../allure-results",
  },
  screenshotOnRunFailure: true,
  reporter: "mocha-allure-reporter",
  defaultCommandTimeout: 20000,
  pageLoadTimeout: 30000,
  screenshotsFolder: "cypress/screenshots",
  videosFolder: "cypress/videos",
  video: false,
  
  retries: {
    runMode: 1,
    openMode: 0,
  },
  allure: {
    // Reuse the existing report folder for retries
    allureReuseExisting: true,
  },
  
  
  e2e: {
    experimentalOriginDependencies: true,
    beforeEach() {
      // Remove the existing Allure report folder
      fs.rmdirSync("allure-results", { recursive: true });
    },
    setupNodeEvents(on, config) {
      const environmentName = config.env.AHCONNECTENV;
      if (!environmentName) {
        throw new Error('Environment name is undefined or null.');
      }

      const environmentFilename = `./${environmentName}.json`;
      const settings = require(environmentFilename);

      if (settings.env) {
        config.env = settings.env;
      }

      on("task", {
        out(message) {
          // Info log in blue colour
          console.log("\x1b[32m%s\x1b[32m", message);
          return null;
        },
        fileExists(filePath) {
          if (fs.existsSync(filePath)) {
            return true;
          } else {
            throw new Error(`File does not exist at path: ${filePath}`);
          }
        },
        allureReportTest(message) {
          // Handle the allureReportTest task here
          // You can access the message passed to the task as `message`
          console.log("Received message from allureReportTest task:", message);
          // Implement your logic to handle the task
          // For example, you can generate the Allure report based on the message
          return null; // Return a value if needed
        },
        
      });
      on('after:run', (results) => {
        const data = `Environment=${environmentName}\nBrowser=${results.browserName}\nBrowserVersion=${results.browserVersion}\nTimestamp=${new Date().toString()}`
        fs.writeFile('allure-results/environment.properties', data, (err) => {
          if (err) throw err;
        });
      })

      allureWriter(on, config);
      allureCypress(on);
      
      return config;
    }
    
  }
});

Attaching the allure report:
![image](https://github.com/allure-framework/allure2/assets/113418492/bb0c7063-3c78-4d10-80ba-e38b0449103c)

Attaching allure-results folder
![image](https://github.com/allure-framework/allure2/assets/113418492/bf3044e0-238e-4265-98be-a07b0b35345f)

**Comment by baev** [comment, marked as answer]
fixed in 3.0.0-beta6
